server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/CompactionFinalizer.java

```diff
@@ -100,8 +102,11 @@ protected CompactionFinalizer(ServerContext context, ScheduledThreadPoolExecutor
     ThreadPools.watchCriticalScheduledTask(schedExecutor.scheduleWithFixedDelay(
         this::notifyTservers, 0, tserverCheckInterval, TimeUnit.MILLISECONDS));
 
-    this.sharedBatchWriter =
-        new SharedBatchWriter(Ample.DataLevel.USER.metaTable(), context, queueSize);
+  }
+
+  private SharedBatchWriter getWriter(ExternalCompactionId ecid) {
+    return writers.computeIfAbsent(ecid.canonical().charAt(0),
```

Comment:
Will have multiple threads executing computeIfAbsent and writers is a hashmap.  Could change it to a concurrent hash map or pre-populate writers in the constructor and only read from the hash map here.

Reply:
lol - I initially had it as a ConcurrentHashMap and thought that I didn't need it.

Reply:
Reverted back to ConcurrentHashMap in 3ebccfc.